src/utils.ts

```diff
@@ -44,11 +52,13 @@ export async function getProjectCompiledSources(rootPath: string, contractsPath:
         remapping: remappings,
         includePath: [rootPath],
     });
-    
+
     return new ASTReader()
         .read(compiledFiles.data, ASTKind.Any, compiledFiles.files)
         // avoid processing files that are not in the specified directory, e.g. node modules or other imported files
-        .filter(sourceUnit => isFileInDirectory(contractsPath, sourceUnit.absolutePath));
+        .filter(sourceUnit => isFileInDirectory(contractsPath, sourceUnit.absolutePath))
+        // avoid processing files from ignored directories
+        .filter(sourceUnit => !ignoredPaths.some(ignoredPath => ignoredPath === sourceUnit.absolutePath));
```

Comment:
can it happen that ignoredPath has an ending `/` and absolute path doesn't and stuff like that?
please check `isFileInDirectory`, I made sure that it is safe to those scenarios